internal/cpp_doc_generator/Doxyfile

```diff
@@ -0,0 +1,10 @@
+@INCLUDE                        = "./Doxyfile.global"
+GENERATE_HTML                   = NO
+GENERATE_XML                    = YES
+XML_PROGRAMLISTING              = NO
+
+PROJECT_NAME                    = "iceoryx2"
+OUTPUT_DIRECTORY                = "./build/doxygen_docs"
```

Comment:
Can this be generated into `target`?

Or is this relative to `cpp_doc_generator`?

Reply:
It is relative to `cpp_doc_generator` but we can put it into `target` if you want.

Reply:
I tried it but it works only partially, some of the folders must be a subfolder of `cpp_doc_generator`. I would leave it as it is for now.

Reply:
What do you think of using `target` instead of `build` within `cpp_doc_generator`.

I'm also wondering whether the generated files will show up in git as untracked files?

Reply:
I would also add it to the git ignore then